### ISSUE TITLE
Edit observing cycle dates often fails

### ISSUE THREAD

**Issue by AllanEngland** [orppst/pst-api-service#62]
Haven't identified a pattern yet, but it regularly fails to update one or more of the dates when the form is submitted.

ProposalManagerView | proposalCycle | dates.tsx

**Comment by AllanEngland** [comment]
All three cycle date modifier endpoints return a 500 server error containing

`com.fasterxml.jackson.databind.JsonMappingException: failed to lazily initialize a collection of role: org.ivoa.dm.proposal.management.Observatory.telescopes: could not initialize proxy - no Session`

The dates are successfully changed in the database, but then building the response (type ProposalCycleDates) fails.

**Comment by AllanEngland** [comment]
To fix the API I simply changed the response type from ProposalCycleDates to a Response containing the new date value.  

When testing it transpired that calling all three updates asynchronously no longer works and (on average) only one of the three updates - presumably transactions are no longer read blocking.
  
So, to fix this I changed the GUI to call the updates synchronously and everything now appears to work again.

This means both API and GUI pull requests are dependent on each other and need to be done at the same time.
